flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/AsyncExecutionController.java

```diff
@@ -220,12 +283,17 @@ <IN, OUT> void insertBlockingBuffer(StateRequest<K, IN, OUT> request) {
      * @param force whether to trigger requests in force.
      */
     void triggerIfNeeded(boolean force) {
-        // TODO: introduce a timeout mechanism for triggering.
         if (!force && stateRequestsBuffer.activeQueueSize() < batchSize) {
+            // if the active queue size is 1, it means that the current request is the oldest one in
+            // the active queue.
+            if (stateRequestsBuffer.activeQueueSize() == 1) {
+                scheduleTimeout(currentTriggerSeq.get());
+            }
```

Comment:
Is it possible trigger multiple times for one seq?

Reply:
No, because the seq number will increase by 1 per trigger.

Reply:
Ah... I've mispoken...  Is it possible `scheduleTimeout` multiple times for one seq?

Reply:
No,  there will definitely be a trigger between two `stateRequestsBuffer.activeQueueSize() == 1`, so when the two `stateRequestsBuffer.activeQueueSize() == 1` conditions are met, seq number must be different.

Reply:
I mean, the `triggerIfNeeded(false)` may be triggered somewhere else right? And even the `enqueueToActive` has two entry. I'd suggest move the `scheduleTimeout` part into the `StateRequestBuffer`, but the handler and seq maintaining part should stay in `AEC`

Reply:
Currently, `triggerIfNeeded(false)` is only called in `handleRequest()`, I added a `schedulingCount==1` condition to avoid "trigger multiple times for one seq" in the future.

And I moved `scheduleTimeout ` into `StateRequestBuffer`, thanks for the suggestion.👍